tests/gpuistl/test_gpuPvt.cu

```diff
@@ -0,0 +1,271 @@
+        OPM_GPU_SAFE_CALL(cudaMalloc(&gpuPressure, sizeof(Evaluation)));
+        OPM_GPU_SAFE_CALL(cudaMemcpy(gpuPressure, &pressure, sizeof(Evaluation), cudaMemcpyHostToDevice));
+    }
+    ~Fixture(){
+        OPM_GPU_SAFE_CALL(cudaFree(gpuTemp));
+        OPM_GPU_SAFE_CALL(cudaFree(gpuPressure));
+    }
+
+    Evaluation temp; // [K]
+    Evaluation pressure; // [Pa]
+    Evaluation* gpuTemp; // [K]
+    Evaluation* gpuPressure; // [Pa]
+
+    double gpuComputedResultOnCpu;
+
+    Opm::CO2Tables<double, std::vector<double>> co2Tables;
+};
+
+// Kernel to evaluate a 2D function on the GPU
+__global__ void gpuEvaluateUniformTabulated2DFunction(GpuTab gpuTab, Evaluation* inputX, Evaluation* inputY, double* result) {
+    *result = gpuTab.eval(*inputX, *inputY, true).value();
+}
+
+// Kernel using a CO2 object on the GPU
+__global__ void gpuCO2GasViscosity(GpuViewCO2Tables gpuViewCo2Tables, Evaluation* temp, Evaluation* pressure, double* result) {
+    *result = GpuCO2::gasViscosity<Evaluation>(gpuViewCo2Tables, *temp, *pressure, true).value();
+}
+
+// Kernel using a SimpleHuDuanH20 object on a GPU
+__global__ void huDuanLiquidDensity(Evaluation* temp, Evaluation* pressure, double* result) {
+    *result = HuDuan::liquidDensity<Evaluation>(*temp, *pressure, true).value();
+}
+
+// Kernel using a BrineDynamic object on a GPU
+__global__ void brineDynamicLiquidEnthalpy(Evaluation* temp, Evaluation* pressure, Evaluation* salinity, double* result) {
+    *result = BrineDyn::liquidEnthalpy<Evaluation>(*temp, *pressure, *salinity).value();
+}
+
+// Kernel using a Brine_CO2 object on a GPU
+__global__ void brineCO2GasDiffCoeff(GpuViewCO2Tables co2tables, Evaluation* temp, Evaluation* pressure, double* result) {
+    *result = GpuBrine_CO2::gasDiffCoeff<Evaluation, GpuViewCO2Tables>(co2tables, *temp, *pressure, true).value();
+}
+
+// Kernel using a Co2GasPvt object on a GPU
+__global__ void co2GasPvtInternalEnergy(GpuViewCo2Pvt gpuViewCo2Pvt, Evaluation* temp, Evaluation* pressure, double* result) {
+    *result = gpuViewCo2Pvt.internalEnergy<Evaluation>(1, *temp, *pressure, Evaluation(0.4), Evaluation(0.0)).value();
+}
+
+// Helper function to launch a kernel and retrieve the result on the CPU to reduce code duplicatoin
+template <typename KernelFunc, typename... Args>
+double launchKernelAndRetrieveResult(KernelFunc kernel, Args... args) {
+    double* resultOnGpu;
+    double gpuComputedResultOnCpu;
+
+    // Allocate memory for the result on the GPU
+    OPM_GPU_SAFE_CALL(cudaMalloc(&resultOnGpu, sizeof(double)));
+
+    // Launch the kernel
+    kernel<<<1, 1>>>(args..., resultOnGpu);
+
+    // Check for any errors in kernel launch
+    OPM_GPU_SAFE_CALL(cudaPeekAtLastError());
+    OPM_GPU_SAFE_CALL(cudaDeviceSynchronize());
+
+    // Retrieve the result from the GPU to the CPU
+    OPM_GPU_SAFE_CALL(cudaMemcpy(&gpuComputedResultOnCpu, resultOnGpu, sizeof(double), cudaMemcpyDeviceToHost));
```

Comment:
What would `gpuComputedResultOnCpu = *resultOnGpu;` do?

Reply:
That would not yield anything meaningful, you cannot dereference a gpu pointer and get the value in CPU code